ksqldb-engine/src/main/java/io/confluent/ksql/function/FunctionLoaderUtils.java

```diff
@@ -178,15 +179,18 @@ static SchemaProvider handleUdfReturnSchema(
       final Map<GenericType, SqlType> genericMapping = new HashMap<>();
       for (int i = 0; i < Math.min(parameters.size(), arguments.size()); i++) {
         final ParamType schema = parameters.get(i);
-
-        // we resolve any variadic as if it were an array so that the type
-        // structure matches the input type
-        final SqlType instance = isVariadic && i == parameters.size() - 1
-            ? SqlTypes.array(arguments.get(i).getSqlType())
-            : arguments.get(i).getSqlType();
-
-        genericMapping.putAll(
-            GenericsUtil.resolveGenerics(schema, SqlArgument.of(instance)));
+        if (schema instanceof LambdaType) {
+          genericMapping.putAll(GenericsUtil.resolveGenerics(schema, arguments.get(i)));
+        } else {
+          // we resolve any variadic as if it were an array so that the type
```

Comment:
we should fail variadic lambdas. I doubt anyone is crazy enough to do this, but let's make sure we're failing it gracefully